<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogEditProfile</class>
 <widget class="QDialog" name="DialogEditProfile">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>802</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Edit</string>
  </property>
  <layout class="QHBoxLayout" name="dialog_layout">
   <item>
    <widget class="QWidget" name="left_w" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>400</width>
       <height>0</height>
      </size>
     </property>
     <layout class="QVBoxLayout" name="left">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Common</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="2" column="1">
            <widget class="MyLineEdit" name="address"/>
           </item>
           <item row="1" column="1">
            <widget class="MyLineEdit" name="name"/>
           </item>
           <item row="3" column="1">
            <widget class="QLineEdit" name="port"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="type_l">
             <property name="text">
              <string>Type</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="port_l">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="address_l">
             <property name="text">
              <string>Address</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Name</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="type"/>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="bean">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string notr="true">Bean</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QWidget" name="fake" native="true"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QGroupBox" name="custom_box">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Custom Outbound Settings</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QPushButton" name="custom_outbound_edit">
             <property name="text">
              <string notr="true">Edit</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="custom_global_box">
          <property name="title">
           <string>Custom Config Settings</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QPushButton" name="custom_config_edit">
             <property name="text">
              <string notr="true">Edit</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QPushButton" name="apply_to_group">
          <property name="text">
           <string>Apply settings to this group</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDialogButtonBox" name="buttonBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="standardButtons">
           <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="right_all_w" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>400</width>
       <height>0</height>
      </size>
     </property>
     <property name="toolTip">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="right_layout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QGroupBox" name="stream_box">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QGridLayout" name="_2">
           <item row="0" column="1">
            <widget class="QComboBox" name="network">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <item>
              <property name="text">
               <string notr="true"/>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">tcp</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">ws</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">http</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">grpc</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">quic</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="packet_encoding">
             <item>
              <property name="text">
               <string notr="true"/>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">packetaddr</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">xudp</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="network_l">
             <property name="toolTip">
              <string>The underlying transport method. It must be consistent with the server, otherwise, the connection cannot be established.</string>
             </property>
             <property name="text">
              <string>Network</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="security">
             <item>
              <property name="text">
               <string notr="true"/>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">tls</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="security_l">
             <property name="toolTip">
              <string>Transport Layer Security. It must be consistent with the server, otherwise, the connection cannot be established.</string>
             </property>
             <property name="text">
              <string>Security</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="packet_encoding_l">
             <property name="toolTip">
              <string>UDP FullCone Packet encoding for implementing features such as UDP FullCone. Server support is required, if the wrong selection is made, the connection cannot be made. Please leave it blank.</string>
             </property>
             <property name="text">
              <string>Packet Encoding</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="multiplex_l">
             <property name="toolTip">
              <string>Server support is required</string>
             </property>
             <property name="text">
              <string>Multiplex</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="multiplex">
             <item>
              <property name="text">
               <string>Keep Default</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>On</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Off</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="network_box">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Network Settings (%1)</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="1" column="0">
          <widget class="QLabel" name="path_l">
           <property name="toolTip">
            <string notr="true">http path (ws/http/伪装http)
serviceName (gRPC)
key (QUIC)</string>
           </property>
           <property name="text">
            <string notr="true">Path</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="MyLineEdit" name="host"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="host_l">
           <property name="toolTip">
            <string notr="true">http host (ws/http/伪装http)
security (QUIC)</string>
           </property>
           <property name="text">
            <string notr="true">Host</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="header_type_l">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string notr="true">伪装头部类型 (tcp/quic)</string>
           </property>
           <property name="text">
            <string notr="true">header</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="MyLineEdit" name="path"/>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="header_type">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <item>
            <property name="text">
             <string notr="true"/>
            </property>
           </item>
           <item>
            <property name="text">
             <string notr="true">http</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="MyLineEdit" name="ws_early_data_length"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="ws_early_data_length_l">
           <property name="text">
            <string notr="true">EarlyData Length</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="MyLineEdit" name="ws_early_data_name"/>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="ws_early_data_name_l">
           <property name="text">
            <string notr="true">EarlyData Name</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="security_box">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>TLS Security Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QCheckBox" name="insecure">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>When enabled, V2Ray will not check the validity of the TLS certificate provided by the remote host (the security is equivalent to plaintext)</string>
             </property>
             <property name="text">
              <string>Allow insecure</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string notr="true"/>
             </property>
             <property name="text">
              <string>Certificate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="certificate_edit">
             <property name="text">
              <string notr="true">Edit</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="_3">
           <item row="0" column="1">
            <widget class="MyLineEdit" name="sni"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_5">
             <property name="toolTip">
              <string>Server name indication, clear text.</string>
             </property>
             <property name="text">
              <string notr="true">SNI</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_8">
             <property name="toolTip">
              <string>Application layer protocol negotiation, clear text. Please separate them with commas.</string>
             </property>
             <property name="text">
              <string notr="true">ALPN</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="MyLineEdit" name="alpn"/>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="tls_camouflage_box">
        <property name="title">
         <string>TLS Camouflage Settings</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="1">
          <widget class="QComboBox" name="utlsFingerprint">
           <property name="editable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string notr="true">Fingerprint</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="reality_pbk_l">
           <property name="toolTip">
            <string>Reality public key. If not empty, turn TLS into REALITY.</string>
           </property>
           <property name="text">
            <string notr="true">Reality Pbk</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="reality_sid_l">
           <property name="toolTip">
            <string>Reality short id. Accept only one value.</string>
           </property>
           <property name="text">
            <string notr="true">Reality Sid</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="MyLineEdit" name="reality_sid"/>
         </item>
         <item row="1" column="1">
          <widget class="MyLineEdit" name="reality_pbk"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="reality_spx_l">
           <property name="toolTip">
            <string notr="true"/>
           </property>
           <property name="text">
            <string notr="true">SpiderX</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="MyLineEdit" name="reality_spx"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyLineEdit</class>
   <extends>QLineEdit</extends>
   <header>ui/widget/MyLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>type</tabstop>
  <tabstop>name</tabstop>
  <tabstop>address</tabstop>
  <tabstop>port</tabstop>
  <tabstop>custom_outbound_edit</tabstop>
  <tabstop>custom_config_edit</tabstop>
  <tabstop>apply_to_group</tabstop>
  <tabstop>network</tabstop>
  <tabstop>security</tabstop>
  <tabstop>packet_encoding</tabstop>
  <tabstop>multiplex</tabstop>
  <tabstop>header_type</tabstop>
  <tabstop>path</tabstop>
  <tabstop>host</tabstop>
  <tabstop>ws_early_data_length</tabstop>
  <tabstop>ws_early_data_name</tabstop>
  <tabstop>insecure</tabstop>
  <tabstop>certificate_edit</tabstop>
  <tabstop>sni</tabstop>
  <tabstop>alpn</tabstop>
  <tabstop>utlsFingerprint</tabstop>
  <tabstop>reality_pbk</tabstop>
  <tabstop>reality_sid</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogEditProfile</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>500</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogEditProfile</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>500</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
